nifi-frontend/src/main/frontend/apps/nifi/src/app/pages/flow-designer/state/flow/flow.effects.ts

```diff
@@ -1556,6 +1569,113 @@ export class FlowEffects {
                             );
                         });
 
+                    this.store
+                        .select(selectProcessor(processorId))
+                        .pipe(
+                            takeUntil(editDialogReference.afterClosed()),
+                            isDefinedAndNotNull(),
+                            filter((processorEntity) => {
+                                return (
+                                    processorEntity.revision.clientId === this.client.getClientId() ||
+                                    (runStatusChanged
+                                        ? false
+                                        : processorEntity.revision.clientId === request.entity.revision.clientId)
+                                );
```

Comment:
This condition can be simplified to

```
return (
    processorEntity.revision.clientId === this.client.getClientId() ||
    (!runStatusChanged &&
        processorEntity.revision.clientId === request.entity.revision.clientId)
);
```

However, this logic isn't quite right. The first condition is intended to allow through updates issued but the current client. Once the dialog is opened, that would only include changes to the scheduled state which impacts the run status.

The second condition is designed to allow any changes issued by the client who made the most recent change as long as the current client hasn't changed the schedule state. The issue I see here is that the other client could have made any modification. For instance, changing a property. The `processorUpdates` in the dialog does not account this. It only considers the status, revision, bulletins, and permissions.

Since the system is currently based on optimistic locking, I think we should ignore changes from other clients like we would for other cases (like regular Processor updates). If another client makes a change, the current client will need to get the most up to date changes by refreshing the canvas. I think we could simple this filter to simply limit to changes made by the current client once they have modified the run status.

```
return (
    runStatusChanged && processorEntity.revision.clientId === this.client.getClientId()
);
```

Reply:
I see. Ok.